docs/ansible-output.md

```diff
@@ -0,0 +1,154 @@
+<!--
+Copyright (c) Ansible Project
+GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+SPDX-License-Identifier: GPL-3.0-or-later
+-->
+
+# Updating ansible-playbook output in RST files
+
+One common problem with Ansible playbook examples in documentation is that while it is helpful to include the playbook's output,
+it is somewhat tedious to update the playbook output, especially when the used plugins or modules change.
+
+Antsibull-docs has a tool, its `antsibull-docs ansible-output` subcommand, that lets you update the `ansible-playbook` output
+in code blocks in RST files, and check whether they need to be updated (for example useful for CI).
+
+## Metadata and code blocks
+
+To know which code blocks to update and what playbook and environment variables to use, you need to provide a `ansible-output-data` directive before the actual code block:
+
+```rst
+.. ansible-output-data::
+
+    env:
+      ANSIBLE_STDOUT_CALLBACK: community.general.tasks_only
+      ANSIBLE_COLLECTIONS_TASKS_ONLY_NUMBER_OF_COLUMNS: "90"
+    playbook: |-
+      - hosts: localhost
+        gather_facts: false
+        tasks:
+          - name: Sort list by version number
+            debug:
+              var: ansible_versions | community.general.version_sort
+            vars:
+              ansible_versions:
+                - '2.8.0'
+                - '2.11.0'
+                - '2.7.0'
+                - '2.10.0'
+                - '2.9.0'
+
+.. code-block:: ansible-output
+
+    TASK [Sort list by version number] ********************************************************
+    ok: [localhost] => {
+        "ansible_versions | community.general.version_sort": [
+            "2.7.0",
+            "2.8.0",
+            "2.9.0",
+            "2.10.0",
+            "2.11.0"
+        ]
+    }
+```
+
+The `ansible-output-data` directive results in no output when the `sphinx_antsibull_ext` Sphinx extension is used, and contains YAML data.
+The `env` dictionary allows to set environment variables that are set when calling `ansible-playbook`.
+In this example, we set an explicit callback stdout plugin (using `ANSIBLE_STDOUT_CALLBACK`),
+and provide configuration for that plugin (`ANSIBLE_COLLECTIONS_TASKS_ONLY_NUMBER_OF_COLUMNS`).
+
+The playbook is provided as a multi-line YAML string.
+Antsibull-docs looks for the next code block with language `ansible-output`, and replaces its contents with the output of `ansible-playbook playbook.yml`,
+where `playbook.yml` is filled with the provided playbook.
```

Comment:
It honestly took me a couple of times to read this and then scroll back up and look through things before it really clicked. I think giving an example of the result (what gets generated for the reader) would be helpful and much more immediate.

Reply:
I tried to address this in ada87a707daa8c7e31c1fc0e5c8e4161c05e5676, PTAL.

Reply:
Looks great!